src/components/Calendar/Filters.tsx

```diff
@@ -35,21 +34,44 @@ const CheckboxAndTitleDiv = styled.div`
     min-width: 300px;
 `;
 
+const SelectAllCheckboxes = styled(Checkbox)``;
+
 const CalendarFilters: React.FC<CalendarFilterAccordionProps> = ({
     allLocations,
     currentLocations,
     editLocations,
 }) => {
-    const onChange = (event: React.ChangeEvent<HTMLInputElement>, location: string): void => {
-        if (event.target.checked) {
-            editLocations([...currentLocations, location]);
-            return;
-        }
-        editLocations(currentLocations.filter((testLocation) => testLocation !== location));
+    const onCollectionCentreSelectionChange = (
+        event: React.ChangeEvent<HTMLInputElement>,
+        location: string
+    ): void => {
+        const newLocations = event.target.checked
+            ? currentLocations.concat(location)
+            : currentLocations.filter((testLocation) => testLocation !== location);
```

Comment:
Having a name that corresponds to the list `currentLocations` is probably easier to read? That might just be me.

```suggestion
            : currentLocations.filter((currentLocation) => currentLocation !== location);
```

Reply:
Changed